internal/controller/prefixclaim_controller.go

```diff
@@ -75,7 +75,49 @@ func (r *PrefixClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	/* 1. check if the matching Prefix object exists */
+	/* 1. compute and assign the parent prefix if required */
+	// The current design will use prefixClaim.Status.ParentPrefix for storing the computed parent prefix, and as the source of truth for future parent prefix references
+	if prefixClaim.Status.ParentPrefix == "" /* parent prefix not yet computed/assigned */ {
+		if prefixClaim.Spec.ParentPrefix != "" {
+			// ParentPrefix takes precedence over ParentPrefixSelector
+			prefixClaim.Status.ParentPrefix = prefixClaim.Spec.ParentPrefix
+		} else if len(prefixClaim.Spec.ParentPrefixSelector) > 0 {
```

Comment:
Correct me if I'm wrong here.

This code snippet will try to get the **first** available prefix in Netbox that matches the following conditions:

- All fields set under `spec.parentPrefixSelector`
- Tenant
- (Minimum) Prefix length

What about `site`? Could it happen that the first matching available prefix is on a site different from the one specified in `spec.site`?

E.g. the first matching available prefix is from site `DataCenter2`, but in your PrefixClaim spec you specified `site: DataCenter1`. Since this value is not used as a filter for the computation of the ParentPrefix, could this be a potential issue?

Reply:
I guess this issue (if so) could be solved by adding `site` as a parameter inside `spec.parentPrefixSelector`. Then the only 'issue' is that it would be redundant to have the site specified in two places.

Reply:
Now the computation is added to `isParentPrefixCandidate`, so all the prefix candidates have taken in `site` as a condition during filtering. 

Do you think this would be enough :)

Reply:
I will add a new ParentPrefixSelector documenting the latest changes up there for you to review. Stayed tuned for a bit!

Reply:
Please see `ParentPrefixSelectorGuide.md`!

Reply:
Thanks for checking @CapiSSS